tests/queries/0_stateless/01490_nullable_string_to_enum.sql

```diff
@@ -0,0 +1,12 @@
+DROP TABLE IF EXISTS t_enum;
+DROP TABLE IF EXISTS t_source;
+
+CREATE TABLE t_enum(x Enum8('hello' = 1, 'world' = 2)) ENGINE = TinyLog;
+CREATE TABLE t_source(x Nullable(String)) ENGINE = TinyLog;
+
+INSERT INTO t_source (x) VALUES ('hello');
+INSERT INTO t_enum(x) SELECT x from t_source WHERE x in ('hello', 'world');
+SELECT * FROM t_enum;
+
+DROP TABLE IF EXISTS t_enum;
+DROP TABLE IF EXISTS t_source;
```

Comment:
Can this test just use `CAST`? i.e.

```sql
SELECT CAST(CAST('Hello' AS Nullable(String)) AS Enum8('Hello' = 1));
```

Also maybe it worth just add one more case to `01402_cast_nullable_string_to_enum`?
Since there are also some other related cases.

Reply:
Your test case looks cleaner. I'll add it while keeping the old one. I didn't find `01402_cast_nullable_string_to_enum` and yeah, these tests should be placed together.